tuning/data/setup_dataprocessor.py

```diff
@@ -107,15 +107,21 @@ def _get_pretokenized_dataset_handlers(data_args, packing, is_eval_tokenized):
 
 
 ### Data format 2
-def _get_dataset_formatting_handlers(data_args, packing):
+def _get_dataset_formatting_handlers(data_args, packing, is_padding_free=False):
 
     if data_args.response_template is None:
         if packing is False:
-            raise ValueError(
-                "Since dataset_text_field or data_formatter_template \
-                is provided and packing is disabled, \
-                needs a corresponding response template for masking"
-            )
+            if is_padding_free:
+                logger.debug(
+                    "Assuming extended pretraining scenario because, packing is false,"
+                    + " padding_free plugin is used and no response template was provided."
```

Comment:
`extended pretraining scenario` is subjective do we want to change it to something else 
`pretraining scenario` or `predict all tokens` not sure though.